app/Http/Controllers/Api/PasswordResetController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Events\UserResetPasswordHasSucceeded;
+use App\Http\Controllers\Controller;
+use App\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Hash;
+use Symfony\Component\HttpFoundation\Response;
+
+class PasswordResetController extends Controller
+{
+    public function update(Request $request)
+    {
+        $request->validate(
+            ['password' => ['required', 'confirmed']]
+        );
+
+        $mobileNumber = Cache::get($request->token);
+        $user = User::firstWhere('mobile_number', $mobileNumber);
```

Comment:
this fetch all the users then just get  the first item in a collection. can you try 
`$user = User::where('mobile_number',$mobileNumber)->first();`

![image](https://user-images.githubusercontent.com/57001550/84798815-fb0a6200-b02d-11ea-93be-c1171350bac6.png)